src/formatDate.js

```diff
@@ -50,7 +50,50 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const newDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    const part = toFormat[i];
+    const index = fromFormat.indexOf(part);
+
+    newDate[i] = dateArray[index];
+
+    if (!newDate[i]) {
+      newDate[i] = getYear(dateArray, fromFormat, toFormat);
+    }
+  }
+
+  return newDate.join(toFormat[3]);
```

Comment:
save separator to variable to make it clear what have you used in this join method